hub/api/dataset.py

```diff
@@ -238,6 +245,18 @@ def read_only(self, value: bool):
             self.storage.disable_readonly()
         self._read_only = value
 
+    def _like_ds(self, like: Union[str, "Dataset"], like_creds: dict = None):
+        if isinstance(like, str):
+            like = Dataset(like, creds=like_creds)
+        for tensor in like.tensors:
+            tensor_meta = like[tensor].meta
+            self.create_tensor(
+                tensor,
+                htype=tensor_meta.htype,
+                dtype=tensor_meta.dtype,
+                sample_compression=tensor_meta.sample_compression,
+            )
```

Comment:
instead of constructing a dataset with another dataset, can you add a non-static method that works like this:

```python
import hub

ds1 = hub.Dataset("hub://dyllan/dataset1")
ds2 = ds1.copy_empty("hub://dyllan/dataset2")  # note that this URI passed in is where a dataset is created
```

this is similar to a constructor, but helps keep our actual constructor clean and this is more obvious to users

Reply:
disregard this

Reply:
you can disregard this request

Reply:
you can disregard this